File: 2020-week10/c++/2020-week10_3.cpp
#include <iostream>
#include <string>

using namespace std;

string alpha[8] = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};

int main()
{

    // input
    string str;
    cin >> str;

    // 검사
    int answer = 0;
    for (int i = 0; i < str.size(); i++)
    {
        switch (str[i])
        {
        case 'c':
        {
            if (str[i + 1] == '=' || str[i + 1] == '-')
            {
                answer++;
                i++;
                break;
            }

            answer++;
            break;
        }

        case 'd':
        {
            if (str[i + 1] == '-')
            {
                answer++;
                i++;
                break;
            }
            if (str[i + 1] == 'z' && str[i + 2] == '=')
            {
                answer++;
                i += 2;
                break;
            }
            answer++;
            break;
        }

        case 'l':
        {
            if (str[i + 1] == 'j')
            {
                answer++;
                i++;
                break;
            }
            answer++;
            break;
        }

        case 'n':
        {
            if (str[i + 1] == 'j')
            {
                answer++;
                i++;
                break;
            }
            answer++;
            break;
        }

        case 's':
        {
            if (str[i + 1] == '=')
            {
                answer++;
                i++;
                break;
            }
            answer++;
            break;
        }

        case 'z':
        {
            if (str[i + 1] == '=')
            {
                answer++;
                i++;
                break;
            }
            answer++;
            break;
        }

        default:
        {
            answer++;
            break;
        }
        }
    }

    cout << answer;
    return 0;
}

File: 2020-week10/java-sy/2020-week10_3.java
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Croatia_2941 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String word = sc.next();
		
		String[] w = word.split("");
		String[] alpha = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};
        List<String> alphaList = Arrays.asList(alpha);

		int index = 0;
		int answer = 0;
		while(true) {
			if(index == w.length) break;
			
			if(index == w.length - 1) {
				answer++;
				break;
			}
			
			if(index == w.length - 2) {
				String t = w[index] + w[index+1];
				if(alphaList.contains(t)) {
					answer++;
					break;
				}
				else {
					answer+=2;
					break;
				}
			}
			
			if(alphaList.contains(w[index]+w[index+1])) {
				answer++;
				index+=2;
			}
			else if(alphaList.contains(w[index]+w[index+1]+w[index+2])) {
				answer++;
				index+=3;
			}
			else {
				answer++;
				index+=1;
			}
		}
		
		System.out.println(answer);
	}
}
